core/src/main/java/io/parsingdata/metal/expression/value/reference/CurrentIteration.java

```diff
@@ -56,8 +57,10 @@ public CurrentIteration(final ValueExpression level) {
     }
 
     @Override
-    public ImmutableList<Optional<Value>> eval(final ParseState parseState, final Encoding encoding) {
-        return ImmutableList.create(getIteration(parseState, encoding));
+    public ImmutableList<Value> eval(final ParseState parseState, final Encoding encoding) {
+        return getIteration(parseState, encoding)
+            .map(ImmutableList::create)
+            .orElseGet(ImmutableList::new);
```

Comment:
How common is this pattern of `.map(ImmutableList::create).orElseGet(ImmutableList::new);`? If it happens very often we could consider extracting a helper method... Although that would mean passing an `Optional`. Nevermind. :grinning: